### ISSUE TITLE
TK700STI no input selection

### ISSUE THREAD

**Issue by pborsiak** [rrooggiieerr/homeassistant-benqprojector#10]
Hi - could you implement below commands for source set up 
<CR>*sour=hdmi#<CR>
<CR>*sour=hdmi2#<CR>
<CR>*sour=hdmi3#<CR>
and way to set particular source at startup ?

Also - entity goes offline just after adding projector to HA

**Comment by pborsiak** [comment]
I have read other posts and have installed suggested v 0.0.8 from beta stream, but cannot connect to the projector using it. version 0.0.7 works fine (but have no input selection)

**Comment by rrooggiieerr** [comment]
Sorry, I probably overlooked your issue.

The commands you mention are implemented, it's probably your projector that's playing up.

Could you enable debugging and see if there's any thing that goes wrong?

**Comment by rrooggiieerr** [comment]
It was available in one of the earliest versions, probably the first.

If you enable debug logging you're find more details on why it's not working

**Comment by rrooggiieerr** [comment]
Is there anything in the log file that can help to resolve the issue?

**Comment by pborsiak** [comment]
We are just after move - will do my best to start connect projector before
Sunday

**Comment by pborsiak** [comment]
And by the way - what version should I use to test? Stable 7 or beta 8?

**Comment by rrooggiieerr** [comment]
Just go for the latest version

**Comment by pborsiak** [comment]
Oh my - there was a lots of development I see - I have stopped where v.007
was stable and v.008 was beta. Have just installed v.014 and try to test
today

**Comment by pborsiak** [comment]
Hi - I have connected projector to Serial and have connection now - projector is detected but still no input selection nor current selected input visible.

**Comment by rrooggiieerr** [comment]
That looks quite allright. So you don't get any input selection if you open the media player entity?

**Comment by pborsiak** [comment]
SHIEEEEEEE....!!!

![image](https://github.com/rrooggiieerr/homeassistant-benqprojector/assets/143878195/98f60e7b-c49f-4e34-a700-7ca5823c440b)

But still HDMI3 missing :)

**Comment by rrooggiieerr** [comment]
Look at that.

To have all the options of your projector supported you can follow the instructions from the [BenQ projector library](https://github.com/rrooggiieerr/benqprojector.py?tab=readme-ov-file#detecting-your-projector-capabilities)

**Comment by rrooggiieerr** [comment]
So you've got it working now over serial connection, does it also work using WiFi?

**Comment by pborsiak** [comment]
My projector have no WiFi (also have no COM-IP) - ATM it is in COM cable
range, so ATM will stay with it.

Regards,
Pawel Borsiak
M: 0411298410


pt., 28 cze 2024 o 11:31 rrooggiieerr ***@***.***> napisał(a):

**Comment by pborsiak** [comment]
> Look at that.
> 
> To have all the options of your projector supported you can follow the instructions from the [BenQ projector library](https://github.com/rrooggiieerr/benqprojector.py?tab=readme-ov-file#detecting-your-projector-capabilities)

Yeah - but how to get that under HAOS - from what I read I can ssh into container to install pip3 but this aneverything that is not in config will be wiped once HAOS update

**Comment by rrooggiieerr** [comment]
You don't do that on HAOS, you can connect your projector to your pc/laptop and run the command from there

**Comment by pborsiak** [comment]
Ah - then I need to connect it to my C.H.I.P. with serial and debian on it
- Sorry - can do it on Friday :/
Regards,
Pawel Borsiak
M: 0411298410


wt., 2 lip 2024 o 15:56 rrooggiieerr ***@***.***> napisał(a):
